### ISSUE TITLE
Project does not build

### ISSUE THREAD

**Issue by Velociround** [mozilla/firefox-ios#662]
I get compiler errors (missing file) when trying to build the project for a real device:

    clang: error: no such file or directory: '/Users/antoniogermano/Work/Mobile/firefox-ios/ThirdParty/sqlcipher/sqlite3.c'
    clang: error: no input files

I have tried to build it exactly as said on the GitHub page, and I'm building using Xcode.

If I try to run on the simulator instead I get a very similar error, just using another location:

    module sqlite3 [system] {
        // the line below causes the error
        header "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS8.3.sdk/usr/include/sqlite3.h"
        link "sqlite3"
        export *
    }

    /Users/antoniogermano/Work/Mobile/firefox-ios/Storage/modules/module.modulemap:2:12: error: header '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS8.3.sdk/usr/include/sqlite3.h' not found
    header "/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS8.3.sdk/usr/include/sqlite3.h"
           ^
    <unknown>:0: error: could not build Objective-C module 'sqlite3'

Did anybody else have this problem?

**Comment by BigBoneDaddy** [comment]
What versions clang and Xcodes?

**Comment by Velociround** [comment]
Xcode Version 6.4 (6E35b)
Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)
Target: x86_64-apple-darwin14.4.0
Thread model: posix

**Comment by reza-ryte-club** [comment]
Try installing Carthage.pkg from https://github.com/Carthage/Carthage/releases 
Then go to your firefox source directory from terminal.
Give the following command:

./checkout.sh

after installing necessary dependencies open the Client.xcodeproj file from finder with XCode.
Then in order to build Command+B
You can run then with Command+R.

**Comment by pbernery** [comment]
Same issue here.

Where does the sqlite3.c file dependency comes from? The [Sqlite Cipher project](https://github.com/sqlcipher/sqlcipher) does not seem to contain any sqlite3.c file nor does the [firefox for iOS repository](https://github.com/mozilla/firefox-ios).

By the way, but this is more related to Carthage, I had to manually fetch dependencies as Carthage segfaults on my machine with the Firefox for iOS Cartfile.

**Comment by pbernery** [comment]
I managed to make it compile: I first opened the sqlcipher Xcode project, built the sqlcipher target then reopened the Firefox for iOS project and it passed.
It seems to be a build dependency failure of Xcode.

**Comment by pbernery** [comment]
I forgot to mention that I used the latest sqlcipher project for that. However, the default configuration seems to not be good for Firefox. I got back to the version delivered with Firefox. I had to build the sqlcipher dependency (amalgamation) manually. In a Terminal, in the ThirdParty/sqlcipher directory, type:

```shell
$ ./configure --enable-tempstore=yes --with-crypto-lib=commoncrypto CFLAGS="-DSQLITE_HAS_CODEC -DSQLITE_TEMP_STORE=2"
$ make sqlite3.c
```
The command is taken from the amalgamation target Run Script build phase of the sqlcipher Xcode project.
